cuDSS/simple_matrix_market/simple_matrix_market.cpp

```diff
@@ -0,0 +1,470 @@
+        mview = CUDSS_MVIEW_FULL;
+    else if (strcmp(argv[3], "lower") == 0)
+        mview = CUDSS_MVIEW_LOWER;
+    else if (strcmp(argv[3], "upper") == 0)
+        mview = CUDSS_MVIEW_UPPER;
+    else
+    {
+        std::cerr << "\033[38;5;196m"
+          << "Error: Invalid matrix view type."
+          << "\033[0m" << std::endl;        
+          return EXIT_FAILURE;
+    }
+
+    if ((mview != CUDSS_MVIEW_FULL)&&(mtype==CUDSS_MTYPE_GENERAL)){
+        std::cerr << "\033[38;5;208m"
+          << "WARNING: you chose a lower/upper view of the matrix but you also specified that it is general (not symmetric).\n"
+          << "If your matrix file is truly non symmetric, half of the elements will not be used, as the lower / upper part of the \n"
+          << "matrix will be mirrored and you will be solving for the wrong matrix. Or the reader will throw an error."
+          << "\033[0m" << std::endl;
+    }
+
+    // Matrix and optional vector filenames
+    const char *matrix_filename = argv[4];
+    const char *vector_filename = (argc > 5) ? argv[5] : nullptr;
+
+    int *csr_offsets_h = NULL;
+    int *csr_columns_h = NULL;
+    double *csr_values_h = NULL;
+    double *x_values_h = NULL, *b_values_h = NULL;
+
+    int *csr_offsets_d = NULL;
+    int *csr_columns_d = NULL;
+    double *csr_values_d = NULL;
+    double *x_values_d = NULL, *b_values_d = NULL;
+
+    /* Read input matrix from file and allocate host memory accordingly */
+    int failed = matrix_reader(matrix_filename, n, nnz, &csr_offsets_h, &csr_columns_h, &csr_values_h, mview);
+    if (failed){
+       std::cerr << "\033[38;5;196m"
+          << "Reader failed."
+          << "\033[0m" << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    printf("---------------------------------------------------------\n");
+    printf("cuDSS example: solving a real linear %dx%d system from file \"%s\"\n", n, n, matrix_filename);
+    printf("---------------------------------------------------------\n");
+
+    /* Allocate host memory for solution x*/
+    x_values_h = (double *)malloc(nrhs * n * sizeof(double));
+
+    /* Allocate host memory for right hand side b and fill it*/
+    /* Read from file if rhs file is provided */
+    if (vector_filename != nullptr)
+    {
+        int failed = rhs_reader(vector_filename, n, &b_values_h);
+        if (failed){
+        std::cerr << "\033[38;5;196m"
+          << "Reader failed."
+          << "\033[0m" << std::endl;        return EXIT_FAILURE;
+    }
+    }
+    else
+    {
+        std::cout << "\033[38;5;208m No rhs file provided, filling b with 1.0 \033[0m\n";
+        b_values_h = (double *)malloc(nrhs * n * sizeof(double));
+        for (int i = 0; i < n; i++)
+        {
+            b_values_h[i] = 1.0;
+        }
+    }
+    if (!csr_offsets_h || !csr_columns_h || !csr_values_h ||
+        !x_values_h || !b_values_h)
+    {
+        printf("Error: host memory allocation failed\n");
+        return -1;
```

Comment:
You are mixing `printf` and `std::cerr` for errors. According to the standard, that works fine, but I prefer code that uses either the C-style `printf` or the C++-style `std::cout` (instead of both).

Since all of our examples use `printf`, I suggest to change all of the `std::cerr` to `printf` or, if you want to print to `stderr`, you can use `fprintf(stderr, "Error: This is wrong\n");`